extensions/sql-migration/src/models/stateMachine.ts

```diff
@@ -291,6 +349,316 @@ export class MigrationStateModel implements Model, vscode.Disposable {
+					if (await classVariable.isWaitingForFirstTimeRefresh()) {
+						await page.refreshSkuRecommendationComponents();	// update timer
+					}
+				}, refreshIntervalInMs);
+			}
+		}
+
+		if (!this._autoRefreshGetSkuRecommendationHandle) {
+			// start one-time timer to get SKU recommendation
+			clearTimeout(this._autoRefreshGetSkuRecommendationHandle);
+			if (this.refreshGetSkuRecommendationFrequency !== -1) {
+				this._autoRefreshGetSkuRecommendationHandle = setTimeout(async function () {
+					await page.startCardLoading();
+					await classVariable.getSkuRecommendations();
+					await page.refreshSkuRecommendationComponents();
+				}, this.refreshGetSkuRecommendationFrequency);
+			}
+		}
+	}
+
+	public async stopPerfDataCollection(): Promise<boolean> {
+		try {
+			const response = await this.migrationService.stopPerfDataCollection();
+			void vscode.window.showInformationMessage(constants.AZURE_RECOMMENDATION_STOP_POPUP);
+
+			this._stopPerfDataCollectionApiResponse = response!;
+			this._perfDataCollectionStopDate = this._stopPerfDataCollectionApiResponse.dateTimeStopped;
+
+			// stop auto refresh
+			clearInterval(this._autoRefreshPerfDataCollectionHandle);
+			clearInterval(this._autoRefreshGetSkuRecommendationHandle);
+		}
+		catch (error) {
+			console.log(error);
+		}
+
+		// Generate telemetry for stop data collection request
+		this.generateStopDataCollectionTelemetry().catch(e => console.error(e));
+		return true;
+	}
+
+	private async generateStopDataCollectionTelemetry(): Promise<void> {
+		try {
+			sendSqlMigrationActionEvent(
+				TelemetryViews.DataCollectionWizard,
+				TelemetryAction.StopDataCollection,
+				{
+					'sessionId': this._sessionId,
+					'timeDataCollectionStopped': this._perfDataCollectionStopDate?.toString() || ''
+				},
+				{}
+			);
+
+		} catch (e) {
+			console.log(e);
+		}
+	}
+
+	public async refreshPerfDataCollection(): Promise<boolean> {
+		try {
+			const response = await this.migrationService.refreshPerfDataCollection(this._perfDataCollectionLastRefreshedDate);
+			this._refreshPerfDataCollectionApiResponse = response!;
+			this._perfDataCollectionLastRefreshedDate = this._refreshPerfDataCollectionApiResponse.refreshTime;
+			this._perfDataCollectionMessages = this._refreshPerfDataCollectionApiResponse.messages;
+			this._perfDataCollectionErrors = this._refreshPerfDataCollectionApiResponse.errors;
+			this._perfDataCollectionIsCollecting = this._refreshPerfDataCollectionApiResponse.isCollecting;
+
+			if (this._perfDataCollectionErrors?.length > 0) {
+				void vscode.window.showInformationMessage(constants.PERF_DATA_COLLECTION_ERROR(this._assessmentApiResponse?.assessmentResult?.name, this._perfDataCollectionErrors));
+			}
+		}
+		catch (error) {
+			console.log(error);
+		}
+
+		return true;
+	}
+
+	public async isWaitingForFirstTimeRefresh(): Promise<boolean> {
+		const elapsedTimeInMins = Math.abs(new Date().getTime() - new Date(this._perfDataCollectionStartDate!).getTime()) / 60000;
```

Comment:
can the 60000 be a named constant for readability

Reply:
Addressing in another PR https://github.com/microsoft/azuredatastudio/pull/18345